src/lib/statistics/statistics_objects/min_max_filter.cpp

```diff
@@ -20,7 +20,7 @@ template <typename T>
 Cardinality MinMaxFilter<T>::estimate_cardinality(const PredicateCondition predicate_condition,
                                                   const AllTypeVariant& variant_value,
                                                   const std::optional<AllTypeVariant>& variant_value2) const {
-  return 0;
+  Fail("MinMaxFilters cannot be used to estimate cardinalities");
```

Comment:
Theoretically, one could come up with some type of estimation (everything outside MinMax is 0, everything inside
is estimated assuming equi-distribution). For that, we would also need the cardinality of the underlying data.
Currently, as MinMaxFilters are on a per-segment basis and estimate_cardinality is called for an entire column,
there is no use for this.

Reply:
I'd be fine with saying that, while the data structures might be similar, they serve a different purpose and, to avoid mixups, are kept separately.

Reply:
I fine with that as well. But still the comment should say "possible, but we just don't do it".

Reply:
RangeFilters could be expressed as a type of histogram. This is not done for two reasons: First, we like to keep
filters and histograms separate, as they serve different purposes. Having histograms both on a per-segment basis
(for membership queries) and on a per-column basis (for cardinality estimations) could lead to confusion. Second,
building such a histogram would be more expensive than build a RangeFilter, as we would have to look at each value
instead of only looking at the distinct values (which is significantly cheaper for dictionary encoding).

Reply:
Ack.